petshop-be/service/src/main/java/com/turkninja/petshop/user/impl/UserServiceImpl.java

```diff
@@ -170,4 +180,13 @@ private void setDefaultUserRole(UserEntity userEntity) {
         }
         userEntity.setRole(userRoleEntity);
     }
+
+    private void setDefaultUserRole(UserEntity userEntity, UserRole role) {
+        String roleName = role.name().toUpperCase(Locale.ROOT);
+        UserRoleEntity userRoleEntity = userRoleRepository.findByName(roleName)
+                .orElseThrow(() -> new ApplicationException(AppMessage.BAD_REQUEST,
```

Comment:
If a user is already signed up and than is tried to be added as an admin user, the application will throw a generic exception due that the database will raise a duplicate record error. Should we keep a generic exception flow or catch the exception and return an `ApplicationException` as I did here?

Reply:
Admin users can't be added with existing regular user email. we should check if there is a record with same email. If there is we should throw application exception.